File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys

input = sys.stdin.readline

n = int(input())
num_list = [int(input()) for _ in range(n)]

stack = []
command = []
current = 1

for num in num_list:
    while current <= num:
        stack.append(current)
        command.append("+")
        current +=1

    if stack[-1] == num:
        stack.pop()
        command.append("-")
    else:
        print("NO")
        exit()

print("\n".join(command))

File: 백준/Silver/1874. 스택 수열/스택 수열.kt

import java.util.Stack

fun main(){
    val br = System.`in`.bufferedReader()
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    val numList = IntArray(n)
    val stack = Stack<Int>()
    var current = 1

    repeat(n) {
        numList[it] = br.readLine().toInt()
    }
    //println(numList.contentToString())
    for(num in numList){
        // current가 더 커지면 진행불가
        while(current <= num){
            stack.push(current)
            sb.append("+\n")
            current++
        }
        if(stack.peek() == num){
            stack.pop()
            sb.append("-\n")
        }else{
            println("NO")
            System.exit(0)
        }
    }
    println(sb)


}